contracts/BatchReveal.sol

```diff
@@ -88,24 +88,125 @@ abstract contract BatchReveal is
-            _revealBatchSize == 0 ||
+            (_revealBatchSize != 0 && (_collectionSize % _revealBatchSize != 0)) ||
             _revealBatchSize > _collectionSize
         ) {
             revert Launchpeg__InvalidBatchRevealSize();
         }
+        // We assume that if the reveal is more than 100 days in the future, that's a mistake
+        // Same if the reveal interval is longer than 10 days
+        if (
+            _revealStartTime > block.timestamp + 8_640_000 ||
+            _revealInterval > 864_000
+        ) {
+            revert Launchpeg__InvalidRevealDates();
+        }
         revealBatchSize = _revealBatchSize;
         collectionSize = _collectionSize;
-        _rangeLength = (_collectionSize / _revealBatchSize) * 2;
         intCollectionSize = int128(int256(_collectionSize));
+        if (_revealBatchSize != 0) {
+            _rangeLength = (_collectionSize / _revealBatchSize) * 2;
+        }
+        revealStartTime = _revealStartTime;
+        revealInterval = _revealInterval;
+    }
+
+    /// @notice Set the reveal batch size. Can only be set after
+    /// batch reveal has been initialized and before a batch has
+    /// been revealed.
+    /// @dev Set to 0 to disable batch reveal
+    /// @param _revealBatchSize New reveal batch size
+    function _setRevealBatchSize(uint256 _revealBatchSize)
+        internal
+        initialized
+        revealNotStarted
+    {
+        if (_revealBatchSize == 0) {
+            _rangeLength = 0;
+        } else {
+            if (
+                collectionSize % _revealBatchSize != 0 ||
+                _revealBatchSize > collectionSize
+            ) {
+                revert Launchpeg__InvalidBatchRevealSize();
+            }
+            _rangeLength = (collectionSize / _revealBatchSize) * 2;
+        }
+        revealBatchSize = _revealBatchSize;
+        emit RevealBatchSizeSet(_revealBatchSize);
+    }
+
+    /// @notice Set the batch reveal start time. Can only be set after
+    /// batch reveal has been initialized and before a batch has
+    /// been revealed.
+    /// @param _revealStartTime New batch reveal start time
+    function _setRevealStartTime(uint256 _revealStartTime)
+        internal
+        initialized
+        revealNotStarted
+    {
+        if (_revealStartTime > block.timestamp + 8_640_000) {
+            revert Launchpeg__InvalidRevealDates();
+        }
+        revealStartTime = _revealStartTime;
+        emit RevealStartTimeSet(_revealStartTime);
+    }
+
+    /// @notice Set the batch reveal interval. Can only be set after
+    /// batch reveal has been initialized and before a batch has
+    /// been revealed.
+    /// @param _revealInterval New batch reveal interval
+    function _setRevealInterval(uint256 _revealInterval)
+        internal
+        initialized
+        revealNotStarted
+    {
+        if (_revealInterval > 864_000) {
+            revert Launchpeg__InvalidRevealDates();
+        }
```

Comment:
Can we keep a comment saying `if the reveal interval is longer than 10 days, we assume _revealInterval is invalid` ?